src/ophys_etl/transforms/sine_dewarp.py

```diff
@@ -0,0 +1,606 @@
+    b: float
+        Information about the rig used for the experiment.
+    Returns
+    -------
+    np.array
+        The data ready to use in the dewarping process
+    np.array
+        The data ready to use in the dewarping process
+    """
+    xindex = np.zeros(256, np.int)
+    xindexB = np.zeros(256, np.float)  # between pixels
+
+    for j in range(0, int(a)):
+        xindex[j] = (
+            j - int(
+                    (b)
+                    * (1.0 - math.sin((j/(a * 3.0) + 1.0/6.0) * 3.14159265))
+                    + 0.5
+                )
+        )
+
+        # TODO: Should these also use int()?
+        xindexB[j] = (
+            (j + 0.5) - (
+                (b) * (1.0 - math.sin(
+                    ((j + 0.5)/(a * 3.0) + 1.0 / 6.0) * 3.14159265
+                ))
+            )
+        )
+
+    return xindex, xindexB
+
+
+def create_xtable(movie: np.ndarray,
+                  aL: float,
+                  aR: float,
+                  bL: float,
+                  bR: float,
+                  noise_reduction: int):
+    """
+    Compute a number of statistics about the images to be dewarped.
+
+    Parameters
+    ----------
+    movie: np.ndarray
+        The movie that is being dewarped.
+    aL: float
+        Information about the rig used for the experiment.
+    aR: float
+        Information about the rig used for the experiment.
+    bL: float
+        Information about the rig used for the experiment.
+    bR: float
+        Information about the rig used for the experiment.
+    noise_reduction: int
+        The noise reduction method that will be used in dewarping.
+    Returns
+    -------
+    dict
+        Various computed information about the video.
+    """
+    # assume flat area aL - (512-aR)
+
+    meanimg = np.mean(movie, axis=0)  # use avgimg to compute mode
+    meanimg = meanimg/256          # less noisy to get mode from 8bit histogram
+    meanimg = meanimg.astype(int)  # less noisy to get mode from 8bit histogram
+
+    # full movie got Memory Error
+    stdvimg = np.std(movie[::8], axis=0)  # use avgimg to compute mode
+    stdvimg = stdvimg/256          # less noisy to get mode from 8bit histogram
+    stdvimg = stdvimg.astype(int)  # less noisy to get mode from 8bit histogram
+
+    # Left side
+    xindexL, xindexLB = get_xindex(aL, bL)
+
+    # Right side
+    xindexR, xindexRB = get_xindex(aR, bR)
+
+    modelist, mean_maxcount = mode(meanimg)
+    mean_modevalue = modelist[0]
+    mean_minvalue = np.min(meanimg)
+
+    # IMPORTANT!: estmated modevalue is from 8bit img, convert back to 16bit
+    mean_modevalue = mean_modevalue * 256
+    mean_minvalue = mean_minvalue * 256
+
+    modelist, stdv_maxcount = mode(stdvimg)
+    stdv_modevalue = modelist[0]
+    stdv_minvalue = np.min(stdvimg)
+
+    # IMPORTANT!: estmated modevalue is from 8bit img, convert back to 16bit
+    stdv_modevalue = stdv_modevalue * 256
+    stdv_minvalue = stdv_minvalue * 256
+
+    j = 0
```

Comment:
these while/else statements are pretty awkward and the variable names `L` and `R` aren't great.